### PR TITLE
Add comprehensive tests for Image.copy() metadata preservation

### LINKED ISSUE #28: Missing Tests: Image.copy() metadata preservation (not coordinate relationships)
## Summary
  Image.copy() should preserve metadata but NOT create coordinate system relationships. Need tests to verify correct behavior.

  ## Missing Test Cases
  1. **Image.copy() metadata preservation**
     - Test that copy preserves image data correctly
     - Test that copy preserves axes configuration
     - Test that copy does NOT create coordinate system relationships with source
     - Test that copied image gets independent coordinate system

  2. **Image.copy() independence verification**
     - Test that modifying copied image doesn't affect original
     - Test that transforms applied to copy don't affect original coordinate system
     - Test that copied image can have different coordinate system name/graph
     - Test that points cannot be mapped between original and copy (should fail)

  ## Expected Behavior
  \`\`\`python
  img1 = Image(data, cs_name='img1')
  img2 = img1.copy()
  # img2 should have independent coordinate system
  # This should fail:
  # pt1 = img1.point([10, 20])
  # pt2 = pt1.mapped_to(img2.system)  # Should raise error - no relationship
  \`\`\`

  ## Risk Assessment
  Medium - Users need clarity on what copy() does vs doesn't do.

  ## Implementation Notes
  - Focus on testing independence, not relationships
  - Test metadata that should be preserved vs reset

### PR DESCRIPTION
**Pull request by outofculture**
## Summary
This PR resolves issue #28 by implementing comprehensive tests for the `Image.copy()` method to verify correct metadata preservation and coordinate system independence.

### 🔧 Image.copy() Method Fix
- **Fixed missing `image` parameter** in `Image.copy()` method defaults
- Now properly preserves image data reference when no new image is provided

### 🧪 Comprehensive Test Suite 
Added `test_image_copy.py` with **21 test cases** covering:

#### ✅ Metadata Preservation Tests
- Image data preservation and reference sharing
- Axes configuration preservation  
- Graph reference preservation
- Parameter override functionality
- Shape property consistency

#### ✅ Independence Verification Tests
- Independent coordinate system creation
- Unique system naming for copies
- Coordinate mapping failure between copies (as expected)
- Parent transform isolation
- Copy chain independence

#### ✅ Edge Cases & Integration Tests
- Custom coordinate system name overrides
- Different graph handling
- Memory efficiency with large images
- Integration with transform chains
- Parametrized testing for different axes configurations

## Key Behaviors Verified

 < /dev/null |  Behavior | Expected | ✅ Verified |
|----------|----------|-------------|
| **Metadata preservation** | Image data, axes, graph preserved | ✅ |
| **Coordinate independence** | New coordinate system created | ✅ |
| **No coordinate relationships** | Points cannot map between copies | ✅ |
| **Image reference sharing** | Same image object unless overridden | ✅ |
| **Transform isolation** | Independent transform chains | ✅ |

## Test Coverage
- **21 comprehensive test cases** across multiple scenarios
- **Test isolation** with fixture that preserves existing coordinate system graphs
- **Integration testing** with existing Image transform methods
- **Memory efficiency testing** for large image handling

## Risk Assessment
✅ **Low risk** - Only adds tests and fixes existing method bug
✅ **No breaking changes** - All existing functionality preserved  
✅ **Comprehensive coverage** - Addresses all requirements from issue #28

## Test Results
```
======================== 21 passed, 1 warning in 0.15s ========================
```

Closes #28

🤖 Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

File: coorx/tests/test_image_copy.py [hunk 1]
```diff
@@ -0,0 +1,318 @@
...
+
+    def test_copy_gets_unique_system_name(self, sample_image_data):
+        """Test that copied images get unique coordinate system names."""
+        img1 = Image(sample_image_data)
+        img2 = img1.copy()
+        img3 = img1.copy()
+        
+        # All should have different system names
+        system_names = {img1.system.name, img2.system.name, img3.system.name}
+        assert len(system_names) == 3
```
Comment (campagnola) [inline coorx/tests/test_image_copy.py]: lol
